File: backend/src/main/java/com/pharmacy/backend/service/OrderService.java
package com.pharmacy.backend.service;

import com.pharmacy.backend.dto.request.OrderFilterRequest;
import com.pharmacy.backend.dto.request.OrderRequest;
import com.pharmacy.backend.dto.response.ApiResponse;
import com.pharmacy.backend.dto.response.OrderDetailResponse;
import com.pharmacy.backend.dto.response.OrderResponse;
import com.pharmacy.backend.dto.response.PageResponse;
import com.pharmacy.backend.entity.Order;
import com.pharmacy.backend.entity.OrderDetail;

import java.util.List;

public interface OrderService {
    ApiResponse<PageResponse<List<OrderResponse>>> getAllOrders(int pageIndex, int pageSize, OrderFilterRequest request);
    ApiResponse<PageResponse<List<OrderResponse>>> getMyOrders(int pageIndex, int pageSize, String status);
    ApiResponse<List<OrderDetailResponse>> getOrderDetail(Long orderId);
    ApiResponse<?> createOrder(OrderRequest request);
    ApiResponse<OrderResponse> changeOrderStatus(Long id, String status);
    ApiResponse<OrderResponse> changePaymentStatus(Long id, String paymentStatus);
    ApiResponse<Long> getTotalOrder();

    ApiResponse<Long> getAllRevenue();

    ApiResponse<List<OrderResponse>> getFiveNewestOrder();
    ApiResponse<Void> cancelOrder(Long id);
}

File: frontend-user/src/services/orderService.js
import api from './api';

export const orderService = {
  createOrder: async (orderData) => {
    const response = await api.post('/orders', orderData);
    return response.data;
  },

  getMyOrders: async (pageIndex = 1, pageSize = 5, status) => {
    const response = await api.get(`/orders/my-orders?pageIndex=${pageIndex}&pageSize=${pageSize}&status=${status}`);
    return response.data;
  },

  getOrderDetail: async (orderId) => {
    const response = await api.get(`/orders/detail/${orderId}`);
    return response.data;
  }
};
